commit 160ce1f1a8fe64b3e2686ae73fbf051ccfe7c7ef
Author: Mohammed Gamal <m.gamal005@gmail.com>
Date:   Wed Aug 4 05:44:24 2010 +0300

    KVM: x86 emulator: Allow accessing IDT via emulator ops
    
    The patch adds a new member get_idt() to x86_emulate_ops.
    It also adds a function to get the idt in order to be used by the emulator.
    
    This is needed for real mode interrupt injection and the emulation of int
    instructions.
    
    Signed-off-by: Mohammed Gamal <m.gamal005@gmail.com>
    Signed-off-by: Avi Kivity <avi@redhat.com>

diff --git a/arch/x86/include/asm/kvm_emulate.h b/arch/x86/include/asm/kvm_emulate.h
index 1e4a72ce301a..1bbf2b6f2a7e 100644
--- a/arch/x86/include/asm/kvm_emulate.h
+++ b/arch/x86/include/asm/kvm_emulate.h
@@ -139,6 +139,7 @@ struct x86_emulate_ops {
 	void (*set_segment_selector)(u16 sel, int seg, struct kvm_vcpu *vcpu);
 	unsigned long (*get_cached_segment_base)(int seg, struct kvm_vcpu *vcpu);
 	void (*get_gdt)(struct desc_ptr *dt, struct kvm_vcpu *vcpu);
+	void (*get_idt)(struct desc_ptr *dt, struct kvm_vcpu *vcpu);
 	ulong (*get_cr)(int cr, struct kvm_vcpu *vcpu);
 	int (*set_cr)(int cr, ulong val, struct kvm_vcpu *vcpu);
 	int (*cpl)(struct kvm_vcpu *vcpu);